### ISSUE TITLE
RFID is being called before new user is committed

### ISSUE THREAD

**Issue by dogshapedangel** [pumpingstationone/WA2AD#6]
It appears that you cannot commit new fields without first saving the user being created.

EX: user does not exist yet, and has an RFID field specified

```Unhandled Exception: System.InvalidOperationException: The Principal object must be persisted before this method can be called.
   at System.DirectoryServices.AccountManagement.Principal.GetUnderlyingObject()
   at WA2AD.ADActions.addOthePager(Principal userPrincipal, String rfidTag) in C:\Users\Mituna\Source\Repos\pumpingstationone\WA2AD\ADActions.cs:line 20
   at WA2AD.ADActions.CreateUser(Member member) in C:\Users\Mituna\Source\Repos\pumpingstationone\WA2AD\ADActions.cs:line 97
   at WA2AD.ADActions.HandleMember(Member member) in C:\Users\Mituna\Source\Repos\pumpingstationone\WA2AD\ADActions.cs:line 200
   at WA2AD.Program..ctor() in C:\Users\Mituna\Source\Repos\pumpingstationone\WA2AD\Program.cs:line 23
   at WA2AD.Program.Main(String[] args) in C:\Users\Mituna\Source\Repos\pumpingstationone\WA2AD\Program.cs:line 31```

**Comment by dogshapedangel** [comment]
moving the RFID code after the save code "fixes" this perhaps? tested it to work...

```           try
            {
                userPrincipal.Save();
                Console.WriteLine("Created a new user for " + member.FirstName + " " + member.LastName);
            }
            catch (Exception e)
            {
                Console.WriteLine("Exception creating user object for " + member.FirstName + " " + member.LastName + " -> " + e);
            }
            // The user may have an RFID tag       
            FieldValue rfidTagFV = getValueForKey(member, "RFID Tag");
            if (rfidTagFV != null && rfidTagFV.ToString().Length > 0)
            {
                string rfidTag = (string)rfidTagFV.Value;
                addOthePager(userPrincipal, rfidTag);
            }```

**Comment by tachoknight** [comment]
Fixed in [4153f788](https://github.com/pumpingstationone/WA2AD/commit/4153f788fe07961214dd8c66bf41443e821385e4)
